### PR TITLE
Move GCP Aggregator to 'preview' network

### PR DESCRIPTION
**Pull request by jpraynaud**
This PR upgrades the Aggregator hosted on GCP cloud so that it uses the `preview` Cardano network instead of the legacy `testnet`:

- [x] Upgrade Cardano node to `1.35.3`
- [x] Parametrize the `docker-compose.yaml` file so that it can be used with different network
- [x] Update the `poolIds` of the Signer nodes (which `poolIds` are selected among the SPOs on the `preview` network) and reduce their number to `2`
- [x] Use `preview` network with `NETWORK=preview` in the `aggregator.tf` file
- [x] Update documentation

After merge, manually duplicate stores data so that we don't need to wait 2 epochs before signing

Relates to #457

### CODE REVIEW

**Comment by ghubertpalo** [comment]
There is a `CardanoNetwork` enum in mithril-common used in `BeaconProvider` & `CliObserver`. IMHO the enum variant `CardanoNetwork::TestNet` should be changed to `Preview` as well.

**Review by Alenar — APPROVED** [review]

**Comment by jpraynaud** [comment]
> There is a `CardanoNetwork` enum in mithril-common used in `BeaconProvider` & `CliObserver`. IMHO the enum variant `CardanoNetwork::TestNet` should be changed to `Preview` as well.

Actually, the parameters used with the commands to the `cardano-cli` will depend on the `CardanoNetwork` type.

IMH it's better to have the `CardanoNetwork::TestNet` handles a `Test Network` which can either be:
* `testnet` with magic id `1097911063`
* `preprod` with magic id `1`
* `preview` with magic id `2`
* `private` with any magic id

I have added a commit that implements this behavior and also enhances the computation of the `CardanoNetwork` from a config in the Signer/Aggregator.

Let me know if that works for you :slightly_smiling_face:

File: mithril-aggregator/config/preview.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+    "cardano_cli_path": "cardano-cli",
+    "cardano_cli_socket_path": "/tmp/cardano.sock",
+    "network": "preview",
+    "network_magic": 2,
```
Comment (abailly-iohk) [inline mithril-aggregator/config/preview.json:5]: suggestion: use one or the other but not both of symbolic name for `preview` and `network_magic`. If there's an internal mapping from symbolic name to magic, you might want to use that.

File: mithril-infra/aggregator.tf [hunk 2]
```diff
@@ -54,7 +54,7 @@ resource "null_resource" "mithril-aggregator" {
 
   provisioner "remote-exec" {
     inline = [
-      "IMAGE_ID=${var.image_id} GOOGLE_APPLICATION_CREDENTIALS_JSON='${var.google_application_credentials_json}' CURRENT_UID=$(id -u) DOCKER_GID=$(getent group docker | cut -d: -f3)  docker-compose -f /home/curry/docker-compose.yaml --profile all up -d"
+      "NETWORK=preview NETWORK_MAGIC=2 IMAGE_ID=${var.image_id} GOOGLE_APPLICATION_CREDENTIALS_JSON='${var.google_application_credentials_json}' CURRENT_UID=$(id -u) DOCKER_GID=$(getent group docker | cut -d: -f3)  docker-compose -f /home/curry/docker-compose.yaml --profile all up -d"
```
Comment (abailly-iohk) [inline mithril-infra/aggregator.tf:57]: Same remark as above

File: mithril-signer/config/preview.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+    "network": "preview",
+    "network_magic": 2,
```
Comment (abailly-iohk) [inline mithril-signer/config/preview.json:3]: ditto

File: mithril-signer/config/testnet.json [hunk 4]
```diff
@@ -1,5 +1,6 @@
 {
     "network": "testnet",
+    "network_magic": 1097911063,
```
Comment (abailly-iohk) [inline mithril-signer/config/testnet.json:3]: ditto

**Review by abailly-iohk — APPROVED** [review]
I am inclined to approve this for the sake of moving fast but I am surprised by how much code is touched by this seemingly simple change :) Perhaps the documentation at least could be made more resilient to change by sticking to defined network identifier (say. `testnet` which is generic enough) and providing a list of available networks somewhere?

**Comment by abailly-iohk** [comment]
Seems like this is just fine, can we merge it? Given the time it can take to register SPOs, the sooner the better

File: docs/docusaurus.config.js [hunk 5]
```diff
@@ -6,175 +6,175 @@ const darkCodeTheme = require('prism-react-renderer/themes/dracula');
 
 /** @type {import('@docusaurus/types').Config} */
 const config = {
-  title: 'Mithril. A complete guide.',
```
[the remark this answers is not in the document]
Comment (jpraynaud) [reply docs/docusaurus.config.js]: It was due to my IDE that handles `4` spaces indentation when this file is `2` spaces indentation.
It has been fixed :+1:

File: mithril-aggregator/config/preview.json (hunk 1 above)
Comment (jpraynaud) [reply mithril-aggregator/config/preview.json]: The known networks (`preview`, `preprod` and `testnet`) are now automatically mapped to their `network_magic`

File: mithril-infra/aggregator.tf (hunk 2 above)
Comment (jpraynaud) [reply mithril-infra/aggregator.tf]: Done

File: mithril-signer/config/preview.json (hunk 3 above)
Comment (jpraynaud) [reply mithril-signer/config/preview.json]: Done

File: mithril-signer/config/testnet.json (hunk 4 above)
Comment (jpraynaud) [reply mithril-signer/config/testnet.json]: Done

**Review by abailly-iohk — APPROVED** [review]

**Comment by abailly-iohk** [comment]
Thanks a lot @jpraynaud. I understand there's manual work to be done for the deployment, definitely can wait tomorrow morning :)

**Pull request merged**